rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -1578,46 +1578,73 @@ void main_window::DecryptSPRXLibraries()
-			"\nAnd is logged with some sceNpDrm* functions when the game/application which owns \"%0\" is running.").arg(qstr(path));
+		const std::string& path = (*vec_modules)[mod_index];
+		const std::string filename = path.substr(path.find_last_of(fs::delim) + 1);
+
+		const QString hint = tr("Hint: KLIC (KLicense key) is a 16-byte long string. (32 hexadecimal characters, can be prefixed with \"KLIC=0x\" from the log message)"
+			"\nAnd is logged with some sceNpDrm* functions when the game/application which owns \"%0\" is running.").arg(qstr(filename));
 
-		if (tried)
+		if (repeat_count >= 2)
 		{
-			gui_log.error("Failed to decrypt %s with specfied KLIC, retrying.\n%s", old_path, sstr(hint));
+			gui_log.error("Failed to decrypt %s with specfied KLIC, retrying.\n%s", path, sstr(hint));
 		}
 
-		input_dialog dlg(32, "", tr("Enter KLIC of %0").arg(qstr(path)),
-			tried ? tr("Decryption failed with provided KLIC.\n%0").arg(hint) : tr("Hexadecimal only."), "00000000000000000000000000000000", this);
+		input_dialog* dlg = new input_dialog(39, "", tr("Enter KLIC of %0").arg(qstr(filename)),
+			repeat_count >= 2 ? tr("Decryption failed with provided KLIC.\n%0").arg(hint) : tr("Hexadecimal value."), "KLIC=0x00000000000000000000000000000000", this);
 
 		QFont mono = QFontDatabase::systemFont(QFontDatabase::FixedFont);
 		mono.setPointSize(8);
-		dlg.set_input_font(mono, true, '0');
-		dlg.set_clear_button_enabled(false);
-		dlg.set_button_enabled(QDialogButtonBox::StandardButton::Ok, false);
-		dlg.set_validator(new QRegularExpressionValidator(QRegularExpression("^[a-fA-F0-9]*$"))); // HEX only
+		dlg->set_input_font(mono, true, '0');
+		dlg->set_clear_button_enabled(false);
+		dlg->set_button_enabled(QDialogButtonBox::StandardButton::Ok, false);
+		dlg->set_validator(new QRegularExpressionValidator(QRegularExpression("^(K(L(I(C(=(0x?)?)?)?)?)?)?[a-fA-F0-9]{0,32}$"))); // HEX only (with additional klic=0x prefix for convenience))
+		dlg->setAttribute(Qt::WA_DeleteOnClose);
 
-		connect(&dlg, &input_dialog::text_changed, &dlg, [&dlg](const QString& text)
+		connect(dlg, &input_dialog::text_changed, dlg, [dlg](const QString& text)
 		{
-			dlg.set_button_enabled(QDialogButtonBox::StandardButton::Ok, text.size() == 32);
+			dlg->set_button_enabled(QDialogButtonBox::StandardButton::Ok, text.size() - (text.indexOf('x') + 1) == 32);
 		});
 
-		if (dlg.exec() == QDialog::Accepted)
+		dlg->show();
+
+		connect(dlg, &QDialog::accepted, this, [this, iterate, dlg, vec_modules, mod_index, klics_storage, repeat_count]()
 		{
-			return sstr(dlg.get_input_text());
-		}
+			std::string text = sstr(dlg->get_input_text());
+
+			if (usz new_index = decrypt_sprx_libraries(*vec_modules, mod_index, text, *klics_storage); new_index < vec_modules->size())
+			{
+				QTimer::singleShot(0, [dlg, iterate, vec_modules, mod_index, klics_storage, repeat_count, new_index]()
```

Comment:
Capturing dlg seems pointless and the shared pointers are already captured in the iterate function
```suggestion
				QTimer::singleShot(0, [iterate, mod_index, repeat_count, new_index]()
```